File: app/src/main/java/com/absinthe/anywhere_/workflow/FlowNode.java
package com.absinthe.anywhere_.workflow;

import androidx.annotation.NonNull;

import com.absinthe.anywhere_.services.IzukoService;
import com.absinthe.anywhere_.utils.manager.Logger;

public class FlowNode {
    public static final int TYPE_ACCESSIBILITY_TEXT = 0;
    public static final int TYPE_ACCESSIBILITY_VIEW_ID = 1;

    private int type;
    private String content;

    public FlowNode(String text, int type) {
        this.content = text;
        this.type = type;
    }

    public void trigger() {
        Logger.d("trigger");
        switch (type) {
            case TYPE_ACCESSIBILITY_TEXT:
                IzukoService.isClicked(false);
                IzukoService.sInstance.clickTextViewByText(content);
                IzukoService.isClicked(true);
                break;
            case TYPE_ACCESSIBILITY_VIEW_ID:
                IzukoService.isClicked(false);
                IzukoService.sInstance.clickTextViewByID(content);
                IzukoService.isClicked(true);
                break;
            default:
        }
    }

    @NonNull
    @Override
    public String toString() {
        return content;
    }
}

File: app/src/main/java/com/absinthe/anywhere_/workflow/FlowNode.kt
package com.absinthe.anywhere_.workflow

import com.absinthe.anywhere_.services.IzukoService
import timber.log.Timber

class FlowNode(private val content: String, private val type: Int) {

    fun trigger() {
        Timber.d("trigger")

        when (type) {
            TYPE_ACCESSIBILITY_TEXT -> {
                IzukoService.getInstance()?.apply {
                    isClicked(false)
                    clickTextViewByText(content)
                    isClicked(true)
                }
            }
            TYPE_ACCESSIBILITY_VIEW_ID -> {
                IzukoService.getInstance()?.apply {
                    isClicked(false)
                    clickTextViewByID(content)
                    isClicked(true)
                }
            }
        }
    }

    override fun toString(): String {
        return content
    }

    companion object {
        const val TYPE_ACCESSIBILITY_TEXT = 0
        const val TYPE_ACCESSIBILITY_VIEW_ID = 1
    }

}